e2e_test/test_fee_currency_fails_on_credit.sh

```diff
@@ -3,10 +3,31 @@
 set -eo pipefail
 
 source shared.sh
+source debug-fee-currency/lib.sh
 
-# Expect that the creditGasFees failed and is logged by geth
-tail -f -n0 geth.log >debug-fee-currency/geth.partial.log & # start log capture
-(cd debug-fee-currency && ./deploy_and_send_tx.sh false true false)
+tail -F -n0 geth.log >debug-fee-currency/geth.partial.log & # start log capture
+(
+	sleep 0.2
+	fee_currency=$(deploy_fee_currency false true false)
+
+	# trigger the first failed call to the CreditFees(), causing the
+	# currency to get temporarily blocklisted.
+	# initial tx should not succeed, should have required a replacement transaction.
+	cip_64_tx $fee_currency | assert_cip_64_tx false
+
+	sleep 2
+
+	# since the fee currency is temporarily blocked,
+	# this should NOT make the transaction execute anymore,
+	# but invalidate the transaction earlier.
+	# initial tx should not succeed, should have required a replacement transaction.
+	cip_64_tx $fee_currency | assert_cip_64_tx false
+
+	cleanup_fee_currency $fee_currency
+)
 sleep 0.5
 kill %1 # stop log capture
-grep "This DebugFeeCurrency always fails in (old) creditGasFees!" debug-fee-currency/geth.partial.log
+# although we sent a transaction wih faulty fee-currency twice,
+# the EVM call should have been executed only once
+grep -a "" debug-fee-currency/geth.partial.log
```

Comment:
Why do we need the `-a`? I don't expect grep to treat this as binary by default.